sdk/src/utils/request.ts

```diff
@@ -8,13 +8,17 @@ import { extractData } from './extract-data.js';
  *
  * @returns The API result if successful
  */
-export const request = async <Output = any>(url: string, options: RequestInit): Promise<Output> => {
+export const request = async <Output = any>(
+	url: string,
+	options: RequestInit,
+	fetcher = globalThis.fetch
```

Comment:
Would it be a possibility to make this a part of the `options` so we don't run into issues with position of the parameters later on?

Reply:
Hmmm interesting, i initially didnt want to do that because options is 1on1 the `fetch` input options so adding an extra property which then needs to be removed before passing it to `fetch` did not seem like the right solution. 🤔

Reply:
How about nesting fetch options inside `options`, something like
```ts
interface Options {
  fetch: typeof globalThis.fetch;
  fetchOptions: RequestInit;
}
```
? 🤔

Reply:
We could do that but im not 100% convinced that that actually improves anything compared to the current approach 🤔

Reply:
Yeah, also not 100% sure, but it would certainly give us some more flexibility because I'm pretty sure sooner or later other options/parameters needs to be added to `request` 🤔😅

Reply:
Discussed on Discord and we both agree that what we currently have makes sense 😄